### PR TITLE
MP-9115 - Factory for converting Primitive types into Nox DateTimeRange

### PR DESCRIPTION
**Pull request by danield-ap**

### CODE REVIEW

File: tests/Nox.Lib.Tests/Factories/Types/NoxTypeDateTimeRangeFactoryTests.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+using Nox.Lib.Tests.FixtureConfig;
+using Nox.Solution;
+
+namespace Nox.Lib.Tests.Factories.Types;
+
+public record DateTimeRangeDto(DateTimeOffset Start, DateTimeOffset End);
+public class NoxTypeDateTimeRangeFactoryTests
+{
+    [Theory, AutoMoqData]
+    public void CreateNoxType_FromDto_IsValid(NoxSolution noxSolution, EntityDefinitionFixture fixture)
```
Comment (dejobratic) [inline tests/Nox.Lib.Tests/Factories/Types/NoxTypeDateTimeRangeFactoryTests.cs:14]: Can we add a test for null a value and an invalid value?

**Review by dejobratic — CHANGES_REQUESTED** [review]
Looks good. We just need to add a few more tests.

File: tests/Nox.Lib.Tests/Factories/Types/NoxTypeDateTimeRangeFactoryTests.cs (hunk 1 above)
Comment (danield-ap) [reply tests/Nox.Lib.Tests/Factories/Types/NoxTypeDateTimeRangeFactoryTests.cs:14]: Yes, tests added.

**Review by gjorgji-grgovski-authoritypartners — APPROVED** [review]

**Review by dejobratic — APPROVED** [review]
